lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyXSDURIResolver.java

```diff
@@ -69,4 +101,48 @@ public Map<String, String> getExternalGrammarLocation(URI fileURI) {
     return externalGrammar;
   }
 
+  /**
+   * Generate the schema file for a LibertyWorkspace using the ws-schemagen.jar in the corresponding Liberty installation
+   * @param libertyWorkspace
+   * @param schemaGenJarPath
+   * @return Path to generated schema file.
+   */
+  private String generateServerSchemaXsd(LibertyWorkspace libertyWorkspace, Path schemaGenJarPath) {
+    //java -jar path/to/ws-schemagen.jar path/to/workspace/.libertyls/libertySchema.xsd
+    File tempDir = LibertyUtils.getTempDir(libertyWorkspace.getWorkspaceURI().toString());
+
+    //TODO: (?) Add subfolders to tempDir: schema/xsd/liberty/server.xsd
+    File xsdDestFile = new File(tempDir, "server.xsd");
```

Comment:
Probably should guard against null here. The `getTempDir()` method has a couple of ways it can return null.

Reply:
Checking if tempDir is null, defaulting to the parent folder of ws-schemagen.jar. Added a null check for the ws-schemagen.jar as well.